CPDShell/Core/algorithms/kliep_cpd.py

```diff
@@ -0,0 +1,75 @@
+from collections.abc import Iterable
+
+import numpy as np
+
+from CPDShell.Core.algorithms.DensityBasedCPD.abstracts import DensityBasedAlgorithm
```

Comment:
IDE: Cannot find reference 'DensityBasedAlgorithm' in 'imported module CPDShell.Core.algorithms.DensityBasedCPD.abstracts'

Does it work on your device? Maybe:
`from CPDShell.Core.algorithms.DensityBasedCPD.abstracts.density_based_algorithm import DensityBasedAlgorithm`